test/test.sh

```diff
@@ -0,0 +1,200 @@
+}
+
+# Subcommand: build.
+build() {
+    local output=
+    local cargo_args=(--cargo-arg=--target=x86_64-unknown-linux-musl)
+
+    while [[ "$#" -gt 0 ]]; do
+        case "$1" in
+        --output)
+            shift
+            output="$1"
+            ;;
+        --cargo-arg)
+            shift
+            cargo_args+=("--cargo-arg=$1")
+            ;;
+        *)
+            echo "Unknown option for build: $1"
+            exit 1
+            ;;
+        esac
+        shift
+    done
+
+    : "${output:=$(aya_tmp)/initrd_image.cpio}"
+
+    # The xtask build-integration-test prints the path to the built image to
+    # stdout. The bluebox command takes arguments for the executable (-e) after
+    # a colon. Use --test-threads=1 to prevent tests from interacting with
+    # shared kernel state due to the lack of inter-test isolation.
+    cargo xtask build-integration-test "${cargo_args[@]}" |
+        xargs -t -I {} bluebox -e {}:'--test-threads=1' -o "$output"
+    echo "Built initrd image to $output"
+}
+
+# Subcommand: download-kernel
+download_kernel() {
+    local version
+    local output
+
+    while [[ "$#" -gt 0 ]]; do
+        case "$1" in
+        --version)
+            shift
+            version="$1"
+            ;;
+        --output)
+            shift
+            output="$1"
+            ;;
+        *)
+            echo "Unknown option for download-kernel: $1"
+            exit 1
+            ;;
+        esac
+        shift
+    done
+
+    if [[ -z ${version:-} ]]; then
+        echo "You must specify a kernel version with --version."
+        exit 1
+    fi
+
+    : "${output:=$(aya_tmp)/linux-$version.bz}"
+
+    url="https://github.com/cilium/ci-kernels/raw/master/linux-${version}.bz"
+    curl -L "$url" -o "$output"
+    echo "Downloaded kernel to $output"
+}
+
+# Subcommand: run
+run() {
+    local initrd
+    local kernel
+
+    while [[ "$#" -gt 0 ]]; do
+        case "$1" in
+        --initrd)
+            shift
+            initrd="$1"
+            ;;
+        --kernel)
+            shift
+            kernel="$1"
+            ;;
+        *)
+            echo "Unknown option for run: $1"
+            exit 1
+            ;;
+        esac
+        shift
+    done
+
+    if [[ -z "${initrd:-}" || -z "${kernel:-}" ]]; then
+        echo "Both --initrd and --kernel options are required for the run command."
+        exit 1
+    fi
+
+    local host_arch
+    host_arch=$(uname -m)
+    if [[ $host_arch == arm64 ]]; then
+        host_arch=aarch64
+    fi
+
+    # At the time of writing all the kernel images we test with are built for x86_64.
+    local guest_arch=x86_64
+
+    local machine=q35
+    if [[ $host_arch == "$guest_arch" ]]; then
+        if [[ $(uname -s) == Darwin ]]; then
+            machine+=',accel=hvf'
+        elif [[ -c /dev/kvm ]]; then
+            machine+=',accel=kvm'
+        fi
+    fi
+
+    log_file=$(mktemp)
+    trap 'rm -rf "$log_file"' EXIT
+
+    qemu-system-"$host_arch" \
```

Comment:
This should be `qemu-system-"$guest_arch"` otherwise on ARM you're trying to run an x86_64 guest with `qemu-system-x86_64` which won't work.